arangod/Cluster/SynchronizeShard.cpp

```diff
@@ -935,7 +935,7 @@ bool SynchronizeShard::first() {
                             leader, lastTick, builder);
       if (!tickResult.ok()) {
         LOG_TOPIC("0a4d4", INFO, Logger::MAINTENANCE) << syncRes.errorMessage();
-        _result.reset(tickResult.result());
+        _result.reset(std::move(tickResult).result());
```

Comment:
are the `)` here correct?
Do you want to move the result() or the tickResult only?
I assume the result should be moved right?

Reply:
Nope, that's correct. There are two methods:
```
  Result const& result() const& { return _result; }
  Result result() && { return std::move(_result); }
```
This way, the latter is called, and the `Result` can be moved out of the `ResultT`.